snuba/subscriptions/scheduler.py

```diff
@@ -321,21 +327,50 @@ def __reset_builder(self) -> None:
             # We are transitioning between jittered and immediate mode. We must use the delegate builder.
             self.__builder = self.__delegate_builder
 
+    def __filter_subscriptions(
+        self, subscriptions: List[Subscription]
+    ) -> List[Subscription]:
+
+        filtered_subscriptions: List[Subscription] = []
+        for subscription in subscriptions:
+            # get the metadata and org_id from the Subscription
+            sub_data = subscription.data
+            sub_metadata = sub_data.metadata
+            org_id = sub_metadata["organization"]
```

Comment:
Is there a way to make access to the metadata fields be specific to an Entity? Not all subscriptions would have an `organization` field. So if tomorrow this method gets called for some other Entity, this would cause an Exception

Reply:
Good point. I can just move the `if` statement up, which checks for generics metrics EntityKeys

Reply:
Good point. Would it make sense to move the `if` statement that checks for generic metrics EntityKeys to `__get_subscriptions` to the filter step? This way, if the corresponding Entity for a subscription is not within generic metrics, we don't do the filter step.

Reply:
Good point. I think can add a condition for checking the EntityKey that corresponds with a Subscription. Where we add  this condition depends:

- if `slice_id` is not None, but the entity of the subscription is not within generic metrics, should we just return the regular list of subscriptions? 

Or, should we just restrict the filter step to Subscriptions that satisfy these conditions: (1) `slice_id` is not None (2) the EntityKey for the Subscription is within generic metrics

Reply:
Lets allow filtering to be called when slice_id is enabled. Within the filter method you can restrict it to specific EntityKey.

Reply:
Ok, that makes sense to me. Just to be clear, does this mean that if we have a `slice_id` enabled, but a non-generic metrics entity, we return all of the subscriptions (no filtering needed) or something else?

Reply:
Yes. The filter can just return the original subscriptions list for non-generic metrics entity

Reply:
Done!